File: android/app/src/main/kotlin/com/transport_system/ts_driver/native_calling_plugin/NativeCallingEvents.kt
package com.transport_system.ts_driver.native_calling_plugin

enum class NativeCallingEvents {
    CALL_RECEIVED,
    CALL_PLACED,
    CALL_TIME,
    CALL_ENDED,
    CALL_DECLINED,
    CALL_NO_ANSWER,
    CALL_USER_BUSY,
    CALL_FAILED;

    fun getName(): String {
        return when (this) {
            CALL_RECEIVED -> "callReceived"
            CALL_PLACED -> "callPlaced"
            CALL_TIME -> "callTime"
            CALL_ENDED -> "callEnded"
            CALL_DECLINED -> "callDeclined"
            CALL_NO_ANSWER -> "callNoAnswer"
            CALL_USER_BUSY -> "callUserBusy"
            CALL_FAILED -> "callFailed"
        }
    }
}

File: ios/Runner/Callkit/Enums/NativeCallingEvents.swift
//
//  NativeCallingEvents.swift
//  Runner
//
//  Created by Hashim Khan on 24/01/2024.
//

public enum NativeCallingEvents {
    case callReceived
    case callPlaced
    case callTime
    case callEnded
    case callDeclined
    case callNoAnswer
    case callUserBusy
    case callFailed
    
    func getName() -> String {
        switch self {
        case .callReceived:
            return "callReceived"
        case .callTime:
            return "callTime"
        case .callEnded:
            return "callEnded"
        case .callPlaced:
            return "callPlaced"
        case .callDeclined:
            return "callDeclined"
        case .callNoAnswer:
            return "callNoAnswer"
        case .callUserBusy:
            return "callUserBusy"
        case .callFailed:
            return "callFailed"
        }
    }
}
